ww3tools/ProcSat_Altimeter.py

```diff
@@ -0,0 +1,296 @@
+     defined in ww3tools.yaml
+    '''
+
+    if wconfig['tspace']==1:
+        smethod="Gridded"
+        hmsg="AODN Altimeter data on model grid."
+    elif wconfig['tspace']==2:
+        smethod="AlongTrack"
+        hmsg="AODN Altimeter data processed along track."
+
+    datein = datetime.utcfromtimestamp(AODN_ALONGTRACK['TIME'][0]).strftime('%Y%m%d%H')
+    datefin = datetime.utcfromtimestamp(AODN_ALONGTRACK['TIME'][-1]).strftime('%Y%m%d%H')
+
+    fname=wconfig['path_out']+"Altimeter"+smethod+"_"+wconfig['ftag']+"_"+altsel+"_"+datein+"to"+datefin
+
+    # Save netcdf
+    ncfile = nc.Dataset(fname+".nc", "w", format=wconfig['fnetcdf']) 
+    ncfile.history=hmsg
+    # create  dimensions.
+    ncfile.createDimension('time' , len(AODN['TIME']))
+    ncfile.createDimension('sname' , 1 )
+    # create variables.
+    vflat = ncfile.createVariable('latitude',np.dtype('float32').char,('time'))
+    vflon = ncfile.createVariable('longitude',np.dtype('float32').char,('time'))
+    vft = ncfile.createVariable('time',np.dtype('float64').char,('time'))
+    vfst = ncfile.createVariable('sat_time',np.dtype('float64').char,('time'))
+    vsname = ncfile.createVariable('sat_name',dtype('a25'),('sname'))
+    vfhs = ncfile.createVariable('hs',np.dtype('float32').char,('time'))
+    vfhscal = ncfile.createVariable('hs_cal',np.dtype('float32').char,('time'))
+    vfwnd = ncfile.createVariable('wsp',np.dtype('float32').char,('time'))
+    vfwndcal = ncfile.createVariable('wsp_cal',np.dtype('float32').char,('time'))
+    # Assign units
+    vft.units = 'seconds since 1970-01-01 00:00:00'
+    vfst.units = 'seconds since 1970-01-01 00:00:00'
+    vflat.units = 'degrees_north' ; vflon.units = 'degrees_east'
+    vfwnd.units = 'm/s' ; vfwndcal.units = 'm/s'
+    vfhs.units = 'm'; vfhscal.units = 'm'
+    # Allocate Data
+    vsname[:] = np.array(altsel).astype('str')
+    vflat[:] = np.array(AODN['LATITUDE']); vflon[:] = np.array(AODN['LONGITUDE'])
+    vft[:] = np.array(AODN['TIME']).astype('double'); vfst[:] = np.array(AODN['SAT_TIME']).astype('double'); 
+    vfhs[:] = np.array(AODN['HS']); vfhscal[:] = np.array(AODN['HS_CAL'])
+    vfwnd[:] = np.array(AODN['WSPD']); vfwndcal[:] = np.array(AODN['WSPD_CAL'])
+    ncfile.close()
+    print(' ')
+    print("netcdf ok: "+fname+".nc")
+
+    del vft, vfst, vflat, vflon, vfhs,vfhscal, vfwnd, vfwndcal
+
+    if len(AODN['TIME'])<10**8 :
+        AODN.iloc[:, 2:] = AODN.iloc[:, 2:].round(3)
+        AODN.to_csv(fname+".csv", index=False)
+
+    print("text file ok: "+fname+".csv")
+
+
+if __name__ == "__main__":
+
+    # start time
+    start = timeit.default_timer()
+    # WW3-tools configuration file
+    wconfig=wread.readconfig('ww3tools.yaml')
+
+    # Default date interval
+    datemin='1985010100'; datemax = datetime.utcnow().strftime("%Y%m%d%H")
+    # Default time step (in seconds) to build the final time array (regular)
+    time_step=float(3600.)
```

Comment:
What about the option of no time averaging?  Can we add that option as well?  I don't see how to select that yet.

Reply:
Thanks.
Yes, I included space as a first criterion because different satellites have different time revolutions around the earth, so the length of the segment would vary depending on the satellite mission, which is not ideal. But this can be an option to include (non-default).
For JASON3, for example, 1h (+-30min) time interval it covers 2.850 Km (487 records). This gives a total of 82 segments of 35km (dlim).

Reply:
@ricampos I guess this is related to the question below - does 0 here mean that we are not averaging for time?